solr/CHANGES.txt

```diff
@@ -127,6 +127,8 @@ Improvements
 * SOLR-17300: Http2SolrClient.Builder.withHttpClient now copies HttpListenerFactory (e.g. for auth, metrics, traces, etc.)
   (Sanjay Dutt, David Smiley)
 
+* SOLR-17321: Remove Deprecated URL and replace it with URI in Preparation for Java 21 (Sanjay Dutt, David Smiley)
```

Comment:
I would put this in "Other"; it's code maintenance in nature.  "Improvement" should be something a user might perceive in some way.